flojoy/flojoy_node_venv.py

```diff
@@ -163,7 +164,7 @@ def _bootstrap_venv(
 
 
 class PipInstallThread(threading.Thread):
-    _bounded_semaphore = threading.BoundedSemaphore(4)
+    _bounded_semaphore = threading.BoundedSemaphore(1)
```

Comment:
Super naive comment @Roulbac: would this solve the problem that you mentioned once?

Is this suggestion bellow correct?
```suggestion
    # Limit to a thread for now not to have the same environment be installed
    # by two threads at once which might cause disruption.
    _bounded_semaphore = threading.BoundedSemaphore(1)
```

Reply:
This was already done in https://github.com/flojoy-ai/python/pull/91 and was initially meant to solve the race condition problem. 

However, this assumes that a single main process executes all the functions (and thus spawns all sub-processes to install pip dependencies),  though I observed that it wasn't the case in studio and that processes were spawned at multiple levels, which didn't keep the semaphore consistent across all of them.

Lastly, the file locks allow us to restrict access to 1 environment 1 at a time, though we can still have later in the future the semaphore's bound >1 to install multiple (different) environments at the same time